runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/Pair.java

```diff
@@ -0,0 +1,30 @@
+package uk.ac.bbsrc.tgac.miso.runscanner;
+
+import java.util.Map.Entry;
+
+public class Pair<K, V> implements Entry<K, V> {
+  private final K key;
+  private final V value;
+
+  public Pair(K key, V value) {
+    super();
+    this.key = key;
+    this.value = value;
+  }
+
+  @Override
+  public K getKey() {
+    return key;
+  }
+
+  @Override
+  public V getValue() {
+    return value;
+  }
+
+  @Override
+  public V setValue(V arg0) {
+    throw new IllegalStateException("Pairs are immutable.");
```

Comment:
```
Pair<File, Run> pair = new Pair(file, run);
run.setName("mutated name"); // pair has been mutated
```
I'm not saying it matters. Just that the message is wrong. You'd have to use defensive copying or something, on unknown types, for Pair to actually be immutable